spec/models/schema_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Schema, type: :model do
+  context 'when created' do
+    subject { described_class.new }
+
+    it 'must have an endpoint and a structure' do
+      expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
+      subject.structure = '{ "field": "value" }'
+      expect { subject.save! }.to raise_error(ActiveRecord::StatementInvalid)
+    end
```

Comment:
In rspec, it is generally frowned upon to have multiple `expect` statements within one test unless it is *absolutely critical* that both pass for the test to succeed.  In this case, I think that you might want to change how this test file is structured.